### PR TITLE
New servant clients

### PR DESCRIPTION
**Pull request by divarvel**

### CODE REVIEW

File: files/servant-client.hs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+type API = BasicAuth "user-management" User :> ToServantApi UsersAPI
+
+data UsersAPI r
+  = UsersAPI
+  { getUsers :: r :- Get '[JSON] [User]
+  , postUser :: r:- ReqBody '[JSON] UserData :> Post '[JSON] NoContent
```
Comment (paulrbr-fl) [inline files/servant-client.hs]: missing space between `r` and `:-` operatior :)

Comment (clementd-fretlink) [reply files/servant-client.hs]: thanks!

File: posts/2019-09-10-a-new-tale-of-servant-clients.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+data UserAPI r
+  = UserAPI
+  { getUser    :: r :- Get '[JSON] User
+  , putUser    :: r :- ReqBody '[JSON] UserData :> Put '[JSON] NoContent
+  , deleteUser :: r :- Delete '[JSON] NoContent
+  }
+  deriving Generic
+```
+
+The magical part here is the `r`. Chosing the `r` allows us to get a record
```
[the remark this answers is not in the document]
Comment (clementd-fretlink) [reply posts/2019-09-10-a-new-tale-of-servant-clients.md]: what other options are you thinking of?

[the remark this answers is not in the document]
Comment (antoine-fl) [reply posts/2019-09-10-a-new-tale-of-servant-clients.md]: /s/Chosing/Choosing

[the remark this answers is not in the document]
Comment (clementd-fretlink) [reply posts/2019-09-10-a-new-tale-of-servant-clients.md]: oh right. thanks!

**Pull request merged**